client/app/telecom/telephony/alias/configuration/changeType/telecom-telephony-alias-configuration-changeType.controller.js

```diff
@@ -1,170 +1,214 @@
-    self.checkServerTasksStatus(bulkResult.success).then(() => {
-      // if one of the promises fails, the failed result won't be store in tasksChecked
-      if (self.successfulTasks.length < bulkResult.success.length) {
-        Toast.warn([$translate.instant('telephony_alias_config_change_type_bulk_server_tasks_some_error')]);
-      }
+    return this.simpleModeOptions.includes(this.selectedFeatureType.id)
+      && !_(this.selectedFeatureType.id).isEqual(this.number.featureType);
+  }
 
-      telephonyBulk.getToastInfos(bulkResult, {
-        fullSuccess: $translate.instant('telephony_alias_config_change_type_bulk_all_success'),
-        partialSuccess: $translate.instant('telephony_alias_config_change_type_bulk_some_success', {
-          count: bulkResult.success.length,
-        }),
-        error: $translate.instant('telephony_alias_config_change_type_bulk_error'),
-      }).forEach((toastInfo) => {
-        Toast[toastInfo.type](toastInfo.message, {
-          hideAfter: null,
-        });
+  confirmChoice() {
+    if (!_(this.number.featureType).isEqual('empty')) {
+      const validChoiceModal = this.$uibModal.open({
+        templateUrl: 'app/telecom/telephony/alias/configuration/changeType/confirm/telecom-telephony-alias-configuration-changeType-confirm.html',
+        controller: 'TelecomTelephonyAliasConfigurationChangeTypeConfirmCtrl',
+        controllerAs: '$ctrl',
+        resolve: {
+          currentFeatureType: () => this.$translate.instant(`telephony_alias_config_change_type_label_${this.number.featureType}`),
+        },
       });
 
-      // force v7 reset cache
-      TelephonyMediator.getAll(true).then(() => {
-        $state.go('telecom.telephony.alias.configuration');
+      validChoiceModal.result.then(() => {
+        this.changeType();
       });
-    }, () => {
-      Toast.error([$translate.instant('telephony_alias_config_change_type_bulk_server_tasks_all_error')]);
-      self.loading.changing = false;
+    } else {
+      this.changeType();
+    }
+  }
+
+  changeType() {
+    this.loading = true;
+
+    this.voipServiceAlias.changeNumberFeatureType({
+      billingAccount: this.billingAccount,
+      serviceName: this.serviceName,
+    }, this.selectedFeatureType.id).then(() => {
+      this.Toast.success(this.$translate.instant('telephony_alias_change_type_ok'));
+      this.OvhApiTelephony.Service().v6().resetCache();
+      this.$state.go('^').then(() => {
+        this.$state.reload();
+      });
+    }).catch((error) => {
+      if (!_(error.type).isEqual('poller')) {
+        this.Toast.error(
+          `${this.$translate.instant('telephony_alias_change_type_ko')} ${_.get(error, 'data.message', '')}`,
+        );
+      }
+    }).finally(() => {
+      this.loading = false;
     });
-  };
+  }
 
-  self.onBulkError = function (error) {
-    Toast.error([$translate.instant('telephony_alias_config_change_type_bulk_on_error'), _.get(error, 'msg.data')].join(' '));
-  };
+  checkServerTasksStatus() {
```

Comment:
For this method, the refactor is just a rewrite of previous code, not a change for the concerned feature (bulk action)